src/zeusops_bot/cogs/zeus_upload.py

```diff
@@ -53,8 +53,12 @@ async def zeus_upload(
     @commands.slash_command(name="zeus-set-mission")
     async def zeus_set_mission(self, ctx: discord.ApplicationContext, filename: str):
         """Activate the given a mission file as a Zeus"""
-        await ctx.respond(f"Setting mission to {filename=}")
-        self.reforger_confgen.zeus_set_mission(filename)
+        try:
+            self.reforger_confgen.zeus_set_mission(filename)
+        except ConfigFileNotFound:
+            await ctx.respond(f"Error: could not find mission '{filename}'")
+        else:
+            await ctx.respond(f"Setting mission to {filename=}")
```

Comment:
Oh heck no!

Try/except/else is a really bad pattern, that is hard to understand, and usually is better addressed as a early-exit solution instead:

```suggestion
        try:
            self.reforger_confgen.zeus_set_mission(filename)
            await ctx.respond(f"Setting mission to {filename=}")
        except ConfigFileNotFound:
            await ctx.respond(f"Error: could not find mission '{filename}'")
```

Reply:
I'd prefer keeping the `await ctx.respond` outside of the `try:` block because it's not the part of the code that raises the exception that we're trying to catch. So could instead report the error in the `catch` block and return afterwards. Or is that unnecessary optimisation in your opinion?